src/components/SafeRedirect.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { InjectAppServices } from '../services/pure-di';
+import RedirectToExternalUrl from './RedirectToExternalUrl';
+
+/**
+ * @param { Object } props
+ * @param { string } props.to
+ * @param { import('../services/pure-di').AppServices } props.dependencies
+ */
+function SafeRedirect({ to, dependencies }) {
+  const isPartialUrl = (url) => {
+    return /^\/[^/]/.test(url);
+  };
+
+  const isWhitelisted = (url) => {
+    const loginWhitelist = [
+      'https://academy.fromdoppler.com/',
+      'http://academy.fromdoppler.com/',
+      'http://prod.doppleracademy.com/',
+      'https://prod.doppleracademy.com/',
+      'https://doppleracademy.com/',
+      'http://doppleracademy.com/',
+    ];
+    return !!loginWhitelist.find((element) => url.startsWith(element));
```

Comment:
What do you think about using this?

```suggestion
    return loginWhitelist.some((element) => url.startsWith(element));
```

Or better yet:

```suggestion
    return loginWhitelist.some(url.startsWith);
```

Reply:
Second approach saddly did not work.

Reply:
Thanks!, when I have time I have to try it in order to understand what I am missing.

Reply:
So will I 😄